workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/EntityGameObjectLinkerSystem.cs

```diff
@@ -7,9 +7,11 @@ public class EntityGameObjectLinkerSystem : ComponentSystem
     {
         public EntityGameObjectLinker Linker;
 
+        [Inject] private WorkerSystem worker;
+
         protected override void OnCreateManager(int capacity)
         {
-            Linker = new EntityGameObjectLinker(World, World.GetExistingManager<WorkerSystem>().LogDispatcher);
+            Linker = new EntityGameObjectLinker(World, worker);
```

Comment:
I don't think `worker` exists at this point.

Reply:
Also why give it a worker and a world?

Reply:
Tested it and the `worker` does exist

as to why both: we don't have access to the `World` from the `WorkerSystem` and we want to store the `World` inside the `SpatialOSComponent`

Reply:
the WorkerSystem could expose its own world (via base class) - or the linker could get the system using the world e.g. the deleted code can be moved within the `EntityGameObjectLinker`

Reply:
the other question is: do we want to expose the world to the mono behaviours? we start mixing ECS and monobehaviours in a confusing way

Reply:
good question, the mixing does not sound nice, I'm OK with either way

Reply:
I think we want to give users the ability to use the ECS we just don't want them to have to use it. Have a reference to the world that they will not normally find is fine. Occasionally you need to get stuff for a hack.

Reply:
We don't want to mix things in places where users then have to use ECS objects in their monobehaviour workflow. If they grab that world object they have gone looking for it.

Reply:
So I think this code is fine.

Reply:
Also I'm not sure what other system you'd write without it that didn't just replace monobehaviours. Something needs to hold a reference to something that can see systems in order for that to work.